test/animation.js

```diff
@@ -243,7 +243,10 @@ exports["Animation -- Servo"] = {
   },
 
   keyframeEasing: function(test) {
-
+    // Don't allow time to advance in this test.  Otherwise it's possible for
+    // the animation to progress between setup and the assertion, causing flaky
+    // failures.
+    this.sandbox.useFakeTimers();
```

Comment:
On a separate branch I added a bunch of logging in this test, and I found that when it fails, the animation's `progress` value has advanced slightly from 0.9 between calling `this.animation.enqueue(segment)` and capturing the return value of `this.animation.calculateProgress(this.animation.playLoop.calledAt)`.  [For example, this run reports progress of 0.902](https://travis-ci.org/code-dot-org/johnny-five/jobs/224152325#L1371).

I eventually realized that this test assumes no significant time will pass in that space - that's why we're manually starting the progress at 0.9 - but on very slow machines that might not be true.  The solution is to use sinon's fake clocks to effectively stop time during this test.  The fake timers are implicitly cleaned up by the sandbox when the test is done.